component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/ActivityDAO.java

```diff
@@ -53,7 +54,42 @@ public interface ActivityDAO extends GenericDAO<ActivityEntity, Long> {
    * @throws ActivityStorageException if has any error
    */
   List<String> getUserIdsActivities(Identity owner, long offset, long limit) throws ActivityStorageException;
-  
+
+  /**
+   * @param activityFilter the activity filter
+   * @param spaceIdentityIds the space identity ids
+   * @param offset the offset index
+   * @param limit the maximum number of ActivityEntity to load
+   * @return the activity entities
+   * @throws ActivityStorageException if has any error
+   */
+  List<Long> getActivityByFilter(ActivityFilter activityFilter,
+                                 List<String> spaceIdentityIds,
+                                 int offset,
+                                 int limit) throws ActivityStorageException;
```

Comment:
Use `default` implementation to avoid compilation errors  when changing existing API ([DEVSTD-23](https://github.com/Meeds-io/meeds/blob/develop/CODING_GUIDELINES.md#devstd-23))